miio/gateway_scripts.py

```diff
@@ -0,0 +1,324 @@
+from json import dumps as dumps_orig
+from random import randint
+
+separators = (",", ":")
+
+
+def dumps(data):
+    return dumps_orig(data, separators=separators)
+
+
+fake_device_id = "120009025"
+fake_device_model = "chuangmi.plug.v3"
+
+
+def sid_to_num(sid):
+    lumi, hex_id = sid.split(".")
+    num_id = int.from_bytes(bytes.fromhex(hex_id), byteorder="big")
+    return str(num_id)[-6:]
+
+
+action_prefix = "x.scene."
+
+# Keeping action script tail decimal int it might be used as index in some db
+action_id = {
+    "move": lambda sid: action_prefix + "1" + sid_to_num(sid),
+    "rotate": lambda sid: action_prefix + "2" + sid_to_num(sid),
+    "singlepress": lambda sid: action_prefix + "3" + sid_to_num(sid),
+    "doublepress": lambda sid: action_prefix + "4" + sid_to_num(sid),
+    "shake": lambda sid: action_prefix + "5" + sid_to_num(sid),
+    "longpress": lambda sid: action_prefix + "6" + sid_to_num(sid),
+    "flip90": lambda sid: action_prefix + "7" + sid_to_num(sid),
+    "flip180": lambda sid: action_prefix + "8" + sid_to_num(sid),
+    "shakeair": lambda sid: action_prefix + "9" + sid_to_num(sid),
+    "taptap": lambda sid: action_prefix + "10" + sid_to_num(sid),
+}
+
+
+def _inflate(
+    action,
+    extra,
+    source_sid,
+    source_model,
+    target_id,
+    target_ip,
+    target_model,
+    target_encoded_token,
+    event=None,
+    command_extra="",
+):
+    if event is None:
+        event = action
+
+    lumi, source_id = source_sid.split(".")
+
+    return [
+        [
+            action_id[action](source_sid),
+            [
+                "1.0",
+                randint(1590161094, 1590162094),
+                [
+                    "0",
+                    {
+                        "did": source_sid,
+                        "extra": extra,
+                        "key": "event." + source_model + "." + event,
+                        "model": source_model,
+                        "src": "device",
+                        "timespan": ["0 0 * * 0,1,2,3,4,5,6", "0 0 * * 0,1,2,3,4,5,6"],
+                        "token": "",
+                    },
+                ],
+                [
```

Comment:
This needs to be done using a container class, otherwise it is way too complicated and hard to interpret. See my comment above, I think some of this data can already be stored in the devices what we have, the event handler (or eventscript? or script? I'm not sure) needs to simply have the potential extra fields (like the `timespan`).